### ISSUE TITLE
build issue with boost (again)

### ISSUE THREAD

**Issue by rmandelb** [GalSim-developers/GalSim#215]
Hi all: I'm writing with yet another build issue related to boost (sorry).  This time it's for a student of mine, @RMebane (Rick), who is trying to build GalSim.  Here's a description of what he's done -
- He first used fink to install scons, fftw3, tmv, and boost.  
- He then installed numpy and pyfits from their websites.  

Then, when running scons in the GalSim directory, it gave an error that it was unable to build against numpy, because numpy was installed for a different version of python.  
- He then installed python, numpy, and pyfits through fink and changed the default version of python to the version fink provided (v.
  2.7.3).  

This worked for building against numpy, but then gave an error message that it was unable to build against Boost.Python.  
- He checked the version of python that boost was installed against using otool, and it said it was installed for v. 2.7.0.  
- He then uninstalled boost and reinstalled it from the boost website, specifying the correct version of python (2.7.3) to use.  This was done using the following command:

./bootstrap.sh --with-python=/sw/include/python2.7

Note that when he calls scons in the GalSim directory, it says

SCons is version 2.0.1 using python version 2.7.3
Python is from /sw/include/python2.7

So it seems that boost SHOULD now be installed against the same python that scons is using.  But when he does

scons BOOST_DIR = /Users/Rick/Desktop/Boost

he gets the following error (all earlier tests pass):

dyld: Library not loaded: libboost_python.dylib
 Referenced from:
/Users/Rick/Desktop/GalSim-developers-GalSim-651eca4/.sconf_temp/conftest_10
 Reason: image not found
sh: line 1: 41400 Trace/BPT trap: 5       .sconf_temp/conftest_10 >
".sconf_temp/conftest_10.out"
Checking if we can build against Boost.Python... no
Cannot run program built with Boost.Python.

However -- libboost_python.dylib is located in BOOST_DIR/lib (i.e., /Users/Rick/Desktop/Boost/lib/) so it's not clear why scons can't find it.  When running otool on the dylib, it still claims to be compiled with python 2.7.0 rather than 2.7.3, so apparently the attempt to install boost against 2.7.3 failed.  

In summary, there are 2 issues:
- failure to install boost against the proper python despite giving the proper command-line option (or at least, what I THINK was the proper command-line option!)
- scons can't even find the libboost_python.dylib despite the fact that it exists in BOOST_DIR/lib.

Help???
